modules/grpc-server/src/protojure/pedestal/core.clj

```diff
@@ -47,6 +48,14 @@
   (let [headers (.getRequestHeaders exchange)]
     (reduce (partial assoc-header headers) {} (.getHeaderNames headers))))
 
+(defn- get-ssl-client-cert
+  "Defensively retrivies SSL client certificate from exchange."
+  [^HttpServerExchange exchange]
+  (try
+    (some-> exchange (.getConnection) (.getSslSessionInfo) (.getPeerCertificateChain))
```

Comment:
Could you do a criterium/quick-bench on this just to see what the impact is?  If it's small, we just go with this as is.  If it's larger, one thing we could do is predicate the query on whether SSL is enabled on the server side.  I doubt it will be large, but this is the fast-path so I want to be aware of any implications.

Reply:
To be clear: I am curious to both the impact of calling the undertow API as well as the try/catch.  

Another thought is that we could offer a flag that lets callers elect to have the SSL info gathered.  OTOH, not sure what the other Pedestal containers do.

Reply:
Makes sense, I was also thinking about using `transient` when optionally extending the request-map. I'll do some benchmarking.